core/src/main/java/org/apache/carbondata/core/view/MVProvider.java

```diff
@@ -74,42 +74,21 @@
 
   private static final String STATUS_FILE_NAME = "mv_status";
 
-  private final String storeLocation;
-
   private final Map<String, SchemaProvider> schemaProviders = new ConcurrentHashMap<>();
 
-  private MVProvider(String storeLocation) {
-    this.storeLocation = storeLocation;
-  }
-
-  public static MVProvider get() {
-    String storeLocation =
-        CarbonProperties.getInstance().getProperty(CarbonCommonConstants.STORE_LOCATION);
-    if (storeLocation == null) {
-      throw new RuntimeException(
-          "Property [" + CarbonCommonConstants.STORE_LOCATION + "] is not set.");
-    }
-    return new MVProvider(storeLocation);
-  }
-
   private static String getSchemaPath(String schemaRoot, String viewName) {
     return schemaRoot + CarbonCommonConstants.FILE_SEPARATOR + "mv_schema." + viewName;
   }
 
-  private SchemaProvider getSchemaProvider(String databaseName) {
+  private SchemaProvider getSchemaProvider(MVManager viewManager, String databaseName) {
     String databaseNameUpper = databaseName.toUpperCase();
     SchemaProvider schemaProvider = this.schemaProviders.get(databaseNameUpper);
     if (schemaProvider == null) {
       synchronized (this.schemaProviders) {
         schemaProvider = this.schemaProviders.get(databaseNameUpper);
         if (schemaProvider == null) {
-          String databaseLocation;
-          if (databaseNameUpper.equalsIgnoreCase(CarbonCommonConstants.DATABASE_DEFAULT_NAME)) {
-            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(this.storeLocation);
-          } else {
-            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(this.storeLocation +
-                CarbonCommonConstants.FILE_SEPARATOR + databaseName + ".db");
-          }
+          final String databaseLocation = FileFactory.getCarbonFile(
```

Comment:
1. assign `viewManager.getDatabaseLocation(databaseName)` to a variable as databaseLocation.
2.assign ` FileFactory.getCarbonFile(databaseLocation)` to a variable and reuse same in if check condition at line number 92

Reply:
OK